src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+public class MyHashMap<K,V> implements MyMap<K,V> {
+    private int defaultCapacity = 16;
+    private float loadFactor = 0.75f;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
+        if (table == null) {
+            table = new Node[defaultCapacity];
+            threshold = (int) (defaultCapacity * loadFactor);
+        }
+        if (size == threshold) {
+            resize();
+        }
+        int hash = (key == null) ? 0 : key.hashCode();
+        int indexNode = hash(key);
+        Node<K,V> newNode = new Node<>(hash, key, value, null);
+        if (table[indexNode] == null) {
+            table[indexNode] = newNode;
+            size++;
+        } else {
+            Node<K,V> oldNode = table[indexNode];
+            while (oldNode != null) {
+                if (key == oldNode.key || key != null && key.equals(oldNode.key)) {
+                    oldNode.value = newNode.value;
+                    return;
+                }
+                if (oldNode.next == null) {
+                    oldNode.next = newNode;
+                    size++;
+                    return;
+                }
+                oldNode = oldNode.next;
+            }
+        }
+    }
+
+    private int hash(K key) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
фактично ми тут не просто хеш ключа рахуємо а одразу його індекс